src/patients/list/ViewPatients.tsx

```diff
@@ -48,22 +58,23 @@ const ViewPatients = () => {
   }, [dispatch, debouncedSearchText])
 
   useEffect(() => {
-    setButtonToolBar([
-      <Button
-        key="newPatientButton"
-        outlined
-        color="success"
-        icon="patient-add"
-        onClick={() => history.push('/patients/new')}
-      >
-        {t('patients.newPatient')}
-      </Button>,
-    ])
-
+    if (patients && patients.length > 0) {
```

Comment:
this `patients` variable represents the current patients in state. 

This could be the default page of patients or a search result. When searching for patients and my search results return 0, I would not expect the new icon to appear. 

I think that we should implement a new function called `count` in `Repository` that returns the number of documents that are available for a type. 

Then here, we use that `count` function to determine if the new icon should display or not.

Reply:
i get it, what i don't get is why do we need to implement `count`? 

it seems like `PatientRepository.findAll()` returns all the existing patients in the db, we can use it as:
1. implement a state called `count`
2. at initial load, setCount(PatientRepository.findAll().length)
3. if (count == 0) show `addNewPations` component, else show the table

but it seems that implementing a `count` in `Repository` seems like a better solution here, since it makes `Repository` more flexible for future issues/enhancements